Update Caffeine to 1.1.2

scolby33 opened the pull request:
Due to changes in Mojave, Caffeine has required an update to work with the Accessibility Security preferences.

The previous maintainer has transferred responsibility to a new author, as mentioned in this Stack Exchange answer: https://apple.stackexchange.com/a/343899/215110

This is my first contribution/modification to a Homebrew .rb file. So I may have left some errors by accident. In particular, I'm concerned about the `appcast` I chose, whether to use an SHA or `:latest` given there isn't a version number in the URL, and the `# ... official ...` comment.

I am happy to make any requested modifications.

<!-- If there’s a checkbox you can’t complete for any reason, that's okay, just explain in detail why you weren’t able to do so. -->

After making all changes to the cask:

- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` reports no offenses.
- [x] The commit message includes the cask’s name and version.
- [x] The submission is for [a stable version](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask) or [documented exception](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#but-there-is-no-stable-version).

vitorgalvao commented:
Welcome, and thank you.

> The previous maintainer has transferred responsibility to a new author

That’s not what the answer says. It says they have provided the source code under and MIT license. That’s not the same as officially passing the torch. The answer does also not prove that what the person is saying is true.

As such, unless the official maintainer says in the official website that they have deprecated their version in favour of this new one, it’s effectively a fork, and needs to be treated as such (i.e. a new cask; check the token reference to see how it needs to be named).

scolby33 commented:
Thank you for the education! I have created a new cask `intellisense-caffeine` and updated the same branch.

I am happy to make a new pull if this one cannot be reopened.

My initial questions remain; I have also added the `conflicts_with cask:` annotation for the original `caffeine`.

<!-- If there’s a checkbox you can’t complete for any reason, that's okay, just explain in detail why you weren’t able to do so. -->

After making all changes to the cask:

- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` reports no offenses.
- [x] The commit message includes the cask’s name and version.
- [x] The submission is for [a stable version](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask) or [documented exception](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#but-there-is-no-stable-version).

Additionally, if **adding a new cask**:

- [x] Named the cask according to the [token reference].
- [x] `brew cask install {{cask_file}}` worked successfully.
- [x] `brew cask uninstall {{cask_file}}` worked successfully.
- [x] Checked there are no [open pull requests] for the same cask.
- [x] Checked the cask was not [already refused].
- [x] Checked the cask is submitted to [the correct repo].

[token reference]: https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md
[open pull requests]: https://github.com/Homebrew/homebrew-cask/pulls
[already refused]: https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues
[the correct repo]: https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask
[version-checksum]: https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/sha256.md#updating-the-sha256

vitorgalvao commented:
Please submit a new PR, yes.

As for the `:latest` question, as long as there is an appcast, cask is to versioned.
